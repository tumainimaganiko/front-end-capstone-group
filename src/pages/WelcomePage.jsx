import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSignInAlt, faUserPlus, faCar } from '@fortawesome/free-solid-svg-icons';
import { TOKENKEY } from '../util/auth';
import '../styles/style.css';

const WelcomePage = () => {
  const [isLoggedIn, setIsLoggedIn] = useState(false);

  const checkUserToken = () => {
    if (!localStorage.getItem(TOKENKEY) || localStorage.getItem(TOKENKEY) === undefined) {
      setIsLoggedIn(false);
    } else {
      setIsLoggedIn(true);
    }
  };

  useEffect(() => checkUserToken(), []);
  return (
    <section>
      <div className=" welcome flex flex-col items-center min-h-screen text-white">
        <div className="mt-14">
          <h1 className="text-4xl sm:text-5xl font-bold mb-4 font-sans text-center">
            Welcome to Car Rental
          </h1>
          <p className="text-sm sm:text-lg mb-32 text-center text-black">
            Find the perfect car for your next adventure
          </p>
        </div>
        <div className="flex flex-col sm:flex-row space-y-4 sm:space-y-0 sm:space-x-6">
          { isLoggedIn ? (
            <Link
              to="/cars"
              className="bg-primary hover:bg-lime-400 hover:text-white text-black px-6 rounded-full p-1 text-sm items-end"
            >
              <FontAwesomeIcon icon={faCar} />
              {' '}
              See Available Cars
            </Link>
          ) : (
            <>
              <Link
                to="/LogIn"
                className="bg-primary hover:bg-lime-400 hover:text-gray-500 text-white px-2 rounded-e-full p-1 md:ps-6 text-sm items-end"
              >
                <FontAwesomeIcon icon={faSignInAlt} />
                {' '}
                Log In
              </Link>
              <Link
                to="/SignUp"
                className="bg-primary hover:bg-lime-400 hover:text-gray-500 text-white px-2 rounded-e-full p-1 md:ps-6 text-sm items-end"
              >
                <FontAwesomeIcon icon={faUserPlus} />
                {' '}
                Sign Up
              </Link>
            </>
          )}
        </div>
      </div>
    </section>
  );
};

export default WelcomePage;
