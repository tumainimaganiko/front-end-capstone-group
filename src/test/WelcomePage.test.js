import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import WelcomePage from '../pages/WelcomePage';

describe('Should render correctly', () => {
  it('It should render a WelcomePage', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <WelcomePage />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
