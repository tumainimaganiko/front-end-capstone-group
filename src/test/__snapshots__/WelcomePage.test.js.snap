// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly It should render a WelcomePage 1`] = `
<section>
  <div
    className=" welcome flex flex-col items-center min-h-screen text-white"
  >
    <div
      className="mt-14"
    >
      <h1
        className="text-4xl sm:text-5xl font-bold mb-4 font-sans text-center"
      >
        Welcome to Car Rental
      </h1>
      <p
        className="text-sm sm:text-lg mb-32 text-center text-black"
      >
        Find the perfect car for your next adventure
      </p>
    </div>
    <div
      className="flex flex-col sm:flex-row space-y-4 sm:space-y-0 sm:space-x-6"
    >
      <a
        className="bg-primary hover:bg-lime-400 hover:text-gray-500 text-white px-2 rounded-e-full p-1 md:ps-6 text-sm items-end"
        href="/LogIn"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-right-to-bracket "
          data-icon="right-to-bracket"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 105.9L340.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L217.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1L32 320c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM352 416l64 0c17.7 0 32-14.3 32-32l0-256c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l64 0c53 0 96 43 96 96l0 256c0 53-43 96-96 96l-64 0c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        Log In
      </a>
      <a
        className="bg-primary hover:bg-lime-400 hover:text-gray-500 text-white px-2 rounded-e-full p-1 md:ps-6 text-sm items-end"
        href="/SignUp"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-user-plus "
          data-icon="user-plus"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304h91.4C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7H29.7C13.3 512 0 498.7 0 482.3zM504 312V248H440c-13.3 0-24-10.7-24-24s10.7-24 24-24h64V136c0-13.3 10.7-24 24-24s24 10.7 24 24v64h64c13.3 0 24 10.7 24 24s-10.7 24-24 24H552v64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        Sign Up
      </a>
    </div>
  </div>
</section>
`;
